### PR TITLE
Expose NSX-T T1 router configuration for NSX ALB

### LINKED ISSUE #1801: Expose NSX ALB's NSX-T Connector to support NSX ALB configure to NSX-T
(This is used to request new product features)

**Describe the feature request**

I'd like NSX ALB in TKG can be configured to use NSX-T connector to configure itself in NSX-T Cloud. 

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

Expose NSX-T T1 router for NSX ALB, so NSX ALB can be configured to connect NSX-T.

### PR DESCRIPTION
**Pull request by XudongLiuHarold**
Signed-off-by: Xudong Liu <xudongl@vmware.com>

### What this PR does / why we need it

Expose NSX-T T1 router configurations for NSX ALB to allow NSX ALB can use this connector to connect NSX-T cloud.

### Which issue(s) this PR fixes

Fixes #1801

### Describe testing done for PR

- bolt testing pipeline passed 
- manually tested it in vsphere and NSX-T integrated environment

### Release note

```release-note
expose nsx-t connector for nsx alb to configure nsx-t cloud.
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by lubronzhan — APPROVED** [review]
/lgtm

File: pkg/v1/providers/config_default.yaml [hunk 1]
```diff
@@ -547,13 +547,14 @@ AVI_DISABLE_INGRESS_CLASS: true
...
 AVI_CA_NAME: avi-controller-ca
+AVI_CONTROLLER_VERSION: 20.1.3
 
 AVI_CONTROLLER:
 AVI_USERNAME: ""
 AVI_PASSWORD: ""
 AVI_CLOUD_NAME:
 AVI_SERVICE_ENGINE_GROUP:
-AVI_CONTROLLER_VERSION: 20.1.3
+AVI_NSXT_T1LR:
```
Comment (vuil) [inline pkg/v1/providers/config_default.yaml:557]: It is not very obvious what is supposed to be supplied for this value, how about adding a comment here?

**Review by anujc25 — APPROVED** [review]

File: pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml [hunk 2]
```diff
@@ -88,6 +88,9 @@ data:
   logLevel: "WARN"
   deleteConfig: "false"
   serviceType:  NodePort
+#@ if data.values.AVI_NSXT_T1LR:
```
Comment (vuil) [inline pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml:91]: Something useful to do would be to update
pkg/v1/providers/tests/clustergen/param_models/cluster_vsphere.model
to introduce the configuration of this configuration variable, like:

AVI_NSXT_T1LR: "", (and some other value that can legitimately me set)

so we can see in the clustergen output the effect of the overlay changes in responding these various values used.

**Review by vuil — COMMENTED** [review]
some suggestions about doing what is necessary to evaluate the changes via the clustergen output.

File: pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml (hunk 2 above)
Comment (XudongLiuHarold) [reply pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml:91]: ack, added it

**Pull request merged**